packages/web-components/fast-foundation/src/data-grid/data-grid-cell.ts

```diff
@@ -0,0 +1,310 @@
+import {
+    attr,
+    FASTElement,
+    observable,
+    html,
+    HTMLView,
+    ViewTemplate,
+} from "@microsoft/fast-element";
+import {
+    keyCodeEnter,
+    keyCodeEscape,
+    keyCodeFunction2,
+} from "@microsoft/fast-web-utilities";
+import { ColumnDefinition } from "./data-grid";
+
+const defaultCellContentsTemplate: ViewTemplate = html<DataGridCell>`
+    <template>
+        ${x =>
+            x.rowData === null ||
+            x.columnDefinition === null ||
+            x.columnDefinition.columnDataKey === null
+                ? null
+                : x.rowData[x.columnDefinition.columnDataKey]}
+    </template>
+`;
+
+const defaultHeaderCellContentsTemplate: ViewTemplate = html<DataGridCell>`
+    <template>
+        ${x =>
+            x.columnDefinition === null
+                ? null
+                : x.columnDefinition.title === undefined
+                ? x.columnDefinition.columnDataKey
+                : x.columnDefinition.title}
+    </template>
+`;
+
+/**
+ * Enumerates possible cell types.
+ *
+ * @public
+ */
+export enum DataGridCellTypes {
+    default = "default",
+    columnHeader = "columnheader",
+}
+
+/**
+ * A Data Grid Cell Custom HTML Element.
+ *
+ * @public
+ * @beta
+ */
+export class DataGridCell extends FASTElement {
+    /**
+     * The type of cell
+     *
+     * @public
+     * @remarks
+     * HTML Attribute: cell-type
+     */
+    @attr({ attribute: "cell-type" })
+    public cellType: DataGridCellTypes;
+    private cellTypeChanged(): void {
+        if ((this as FASTElement).$fastController.isConnected) {
```

Comment:
You shouldn't need to `this as FASTElement` here - I think something in your environment is messed up.

Reply:
yeah, I see that, my VS does the right thing though.